homeassistant/components/rfxtrx/__init__.py

```diff
@@ -303,6 +304,7 @@ def _add_device(event, device_id):
         config[CONF_DEVICE_ID] = device_id
 
         data = entry.data.copy()
+        data[CONF_DEVICES] = copy.deepcopy(entry.data[CONF_DEVICES])
```

Comment:
@elupus This was the culprit (or at least one of the). It's never triggering a save with automatic add because it's not making a deepcopy of devices. As a result, `async_update_entry` will never trigger a save. I ran into this problem before for import which was fixed here #39625
I think the scenario still exists when a device was added in 0.113 so it's safe to keep the check.

Reply:
Right. As we are ending up modifying the dict the save will compare against, so it sees no change.

Would be nice if the data returned from the config entry was immutable , so such things would be blocked.